ui: application_name with forward slash breaks stmts page

vladlos opened the pull request:
Replaced `application_name` if have any slashes for fix redirecting to `Statement Details` page.

Added test for encoding and decoding urlSlash

Resolves: #46592

Release note (ui): fix statements page crash on db names with /

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/47423)
<!-- Reviewable:end -->

dhartunian commented:
@elkmaster This change appears too narrowly scoped, can we just urlencode the entire app name? What about a name with two slashes?

elkmaster commented:
@dhartunian, updated util to handle multi slash. Main problem only on routes level. We also have there `databaseNameAttr`, `statementAttr`, `nodeIDAttr`, `tableNameAttr`. Maybe we can proceed with this pr as hotfix, and than create ticket that will cover all cases?

dhartunian commented:
@elkmaster what I had in mind was using the browser API's `encodeURIComponent` function. Is there a reason it's not appropriate here?

dhartunian reviewed (changes requested):
@elkmaster what I had in mind was using the browser API's encodeURIComponent function. Is there a reason it's not appropriate here?

vladlos committed:
ui: application_name with forward slash breaks stmts page

url-encode `application_name` to fix routing problems with app names that contains /

Resolves: #46592

Release note (ui): fix statements page crash on db names with /

elkmaster commented:
@dhartunian, no.

dhartunian commented:
bors r+
